tests/bfd/test_bfd.py

```diff
@@ -379,53 +387,84 @@ def verify_bfd_queue_counters(duthost, dut_intf):
 
 @pytest.mark.parametrize('dut_init_first', [True, False], ids=['dut_init_first', 'ptf_init_first'])
 @pytest.mark.parametrize('ipv6', [False, True], ids=['ipv4', 'ipv6'])
-def test_bfd_basic(request, rand_selected_dut, ptfhost, tbinfo, ipv6, dut_init_first):
+def test_bfd_basic(request, state_db_connection,
+                   rand_selected_dut, ptfhost, tbinfo, ipv6, dut_init_first):
     duthost = rand_selected_dut
     bfd_session_cnt = int(request.config.getoption('--num_sessions'))
     local_addrs, prefix_len, neighbor_addrs, neighbor_devs, neighbor_interfaces = get_neighbors(duthost, tbinfo, ipv6,
                                                                                                 count=bfd_session_cnt)
+
+    prefix = 'BFD_SESSION_TABLE|default|default|'
+    key_pattern = f'{prefix}*'
+    q, ctx = start_subscribe(redis_conn=state_db_connection, key_pattern=key_pattern)
+
     try:
         add_dut_ip(duthost, neighbor_devs, local_addrs, prefix_len)
         init_ptf_bfd(ptfhost)
         add_ipaddr(ptfhost, neighbor_addrs, prefix_len, neighbor_interfaces, ipv6)
         create_bfd_sessions(ptfhost, duthost, local_addrs, neighbor_addrs, dut_init_first)
-
-        time.sleep(1)
+        # check all STATE_DB BFD_SESSION_TABLE neighbors' state is Up
+        status, actual_wait = wait_until_condition(q, prefix, neighbor_addrs,
```

Comment:
If exception is raised in `wait_until_condition`, it would return None. Can this tuple assignment pass if `wait_until_condition` returns `None`? If not, then there is a problem here. And other `wait_until_*` tools have the similar problem.

Reply:
Fixed. `ThreadPoolExecutor.submit` raises `RuntimeError`, `TypeError` and `BrokenThreadPool` all of these shouldn't be handled. The `concurrent.futures.Future.result` `submit()` can raise `TimeoutError`, `CancelledError` and `Exception` which have been handled by logging and returning appropriate return values.